v2/internal/controllers/cdn_profile_afd_test.go

```diff
@@ -0,0 +1,350 @@
+		Spec: cdn.Profile_Spec{
+			Location: to.Ptr("Global"),
+			Owner:    testcommon.AsOwner(rg),
+			Sku:      &cdn.Sku{Name: &sku},
+		},
+	}
+
+	// TODO: See what comes back on the profile based on the child resources being created...
+
+	// Difficult to create these in subtests given the diamond dependency for routes and security policies,
+	// so instead we test them all here.
+	endpoint := newAFDEndpoint(tc, profile)
+	originGroup := newAFDOriginGroup(tc, profile)
+	origin := newAFDOrigin(tc, originGroup)
+	ruleSet := newRuleSet(tc, profile)
+	rule := newRule(tc, ruleSet)
+	route := newRoute(tc, endpoint, originGroup, ruleSet)
+	customDomain := newAFDCustomDomain(tc, profile)
+	firewall := newAFDFirewall(tc, rg)
+	securityPolicy := newSecurityPolicy(tc, profile, endpoint, firewall)
+	// Can't easily test secret as it requires creation of a keyvault secret, but the resource itself is quite simple
+	// so it seems safe to exclude from testing here
+
+	tc.CreateResourcesAndWait(profile, endpoint, originGroup, origin, route, customDomain, firewall, securityPolicy, ruleSet, rule)
+
+	tc.Expect(*profile.Status.Location).To(Equal("Global"))
+	tc.Expect(string(*profile.Status.Sku.Name)).To(Equal("Standard_AzureFrontDoor"))
+
+	tc.RunParallelSubtests(
+		testcommon.Subtest{
+			Name: "CDN Profile Update",
+			Test: func(tc *testcommon.KubePerTestContext) {
+				CDN_Profile_Update(tc, profile)
+			},
+		},
+		testcommon.Subtest{
+			Name: "CDN AfdEndpoint Update",
+			Test: func(tc *testcommon.KubePerTestContext) {
+				CDN_AfdEndpoint_Update(tc, endpoint)
+			},
+		},
+		testcommon.Subtest{
+			Name: "CDN AfdOriginGroup Update",
+			Test: func(tc *testcommon.KubePerTestContext) {
+				CDN_AfdOriginGroup_Update(tc, originGroup)
+			},
+		},
+		testcommon.Subtest{
+			Name: "CDN AfdOrigin Update",
+			Test: func(tc *testcommon.KubePerTestContext) {
+				CDN_AfdOrigin_Update(tc, origin)
+			},
+		},
+		testcommon.Subtest{
+			Name: "CDN Rule Update",
+			Test: func(tc *testcommon.KubePerTestContext) {
+				CDN_Rule_Update(tc, rule)
+			},
+		},
+		testcommon.Subtest{
+			Name: "CDN Route Update",
+			Test: func(tc *testcommon.KubePerTestContext) {
+				CDN_Route_Update(tc, route)
+			},
+		},
+	)
+
+	// Unroll the resources in reverse order to ensure that delete works
+	expectCDNDeleteSucceeds(tc, route, *route.Status.Id)
+	expectCDNDeleteSucceeds(tc, origin, *origin.Status.Id)
+	expectCDNDeleteSucceeds(tc, originGroup, *originGroup.Status.Id)
+	expectCDNDeleteSucceeds(tc, rule, *rule.Status.Id)
+	expectCDNDeleteSucceeds(tc, ruleSet, *ruleSet.Status.Id)
+	expectCDNDeleteSucceeds(tc, securityPolicy, *securityPolicy.Status.Id)
+	expectCDNDeleteSucceeds(tc, endpoint, *endpoint.Status.Id)
+	expectCDNDeleteSucceeds(tc, customDomain, *customDomain.Status.Id)
+
+	expectCDNDeleteSucceeds(tc, profile, *profile.Status.Id)
+}
+
+func expectCDNDeleteSucceeds(tc *testcommon.KubePerTestContext, obj genruntime.ARMMetaObject, id string) {
```

Comment:
Would be good to add this in `testcommon` for future use as well. 
`func ensureDeleteSucceeds(tc *testcommon.KubePerTestContext, obj genruntime.ARMMetaObject, id string, apiVersion string)`

Reply:
Fair point - will add